### PR TITLE
Add ordered argument to Choice HP

### PR DESCRIPTION
**Pull request by omalleyt12**
Allows distinction b/t discrete and categorical values

### CODE REVIEW

File: kerastuner/engine/hyperparameters.py [hunk 1]
```diff
@@ -58,25 +58,31 @@ class Choice(HyperParameter):
...
 
-    def __init__(self, name, values, default=None):
+    def __init__(self, name, values, ordered=None, default=None):
         super(Choice, self).__init__(name=name, default=default)
         if not values:
             raise ValueError('`values` must be provided.')
         self.values = values
+        self.ordered = ordered
+        if self.ordered is None:
+            self.ordered = isinstance(values[0], (int, float))
```
Comment (fchollet) [inline kerastuner/engine/hyperparameters.py]: We should rather do this check for every value in `values`, or alternatively raise a TypeError if there is more than one type in `values`

Comment (omalleyt12) [reply kerastuner/engine/hyperparameters.py]: Done, adding a type check for int, float, str, or bool as well

**Comment by omalleyt12** [comment]
Thanks for the review! Suggested changes are made, PTAL

File: kerastuner/engine/hyperparameters.py [hunk 2]
```diff
@@ -56,27 +56,53 @@ class Choice(HyperParameter):
...
+        types = set(type(v) for v in values)
+        unsupported_types = types - {int, float, str, bool}
+        if unsupported_types:
+            raise TypeError(
+                'A `Choice` can contain only `int`, `float`, '
+                '`str`, or `bool`, found: {}'.format(unsupported_types))
+        if len(types) > 1:
+            raise TypeError(
+                'A `Choice` can contain only one type of value, '
+                'found: {}'.format(types))
```
Comment (fchollet) [inline kerastuner/engine/hyperparameters.py]: Please also print the list of values passed by the user

File: kerastuner/engine/hyperparameters.py [hunk 3]
```diff
@@ -56,27 +56,53 @@ class Choice(HyperParameter):
...
             raise ValueError('`values` must be provided.')
         self.values = values
+
+        # Type checking.
+        types = set(type(v) for v in values)
+        unsupported_types = types - {int, float, str, bool}
+        if unsupported_types:
+            raise TypeError(
+                'A `Choice` can contain only `int`, `float`, '
+                '`str`, or `bool`, found: {}'.format(unsupported_types))
```
Comment (fchollet) [inline kerastuner/engine/hyperparameters.py]: Please also print the list of values passed by the user

Comment (omalleyt12) [reply kerastuner/engine/hyperparameters.py]: Done

File: kerastuner/engine/hyperparameters.py (hunk 2 above)
Comment (omalleyt12) [reply kerastuner/engine/hyperparameters.py]: Done

**Comment by fchollet** [comment]
Please rebase from master.

**Comment by omalleyt12** [comment]
Thanks, done!

**Pull request merged**